distributed/scheduler.py

```diff
@@ -2082,7 +2083,11 @@ def _transition_released_waiting(self, key: Key, stimulus_id: str) -> RecsMsgs:
             assert not ts.who_has
             assert not ts.processing_on
             for dts in ts.dependencies:
-                assert dts.state not in {"forgotten", "erred"}
+                assert dts.state not in {"forgotten", "erred"}, (
+                    ts,
+                    dts,
```

Comment:
```suggestion
                    str(ts),
                    str(dts),
```
otherwise by the time pytest or logging calls `TaskState.__str__` the task can have already changed state

Reply:
Good catch!